### PR TITLE
Refactor A8 AssertionNetworks to enum

### LINKED ISSUE #1511: refactoring AssertionNetworks from array to enum
https://github.com/litentry/litentry-parachain/blob/c73cdc19a575039e8f4bd08e5cf73375635da64b/primitives/core/src/assertion.rs#L47

https://github.com/litentry/litentry-parachain/blob/c73cdc19a575039e8f4bd08e5cf73375635da64b/primitives/core/src/assertion.rs#L28-L29

### PR DESCRIPTION
**Pull request by outofxxx**
resolved #1511

### CODE REVIEW

File: primitives/core/src/assertion.rs [hunk 1]
```diff
@@ -24,8 +24,18 @@ use sp_runtime::{traits::ConstU32, BoundedVec};
 
 type MaxStringLength = ConstU32<64>;
 pub type ParameterString = BoundedVec<u8, MaxStringLength>;
-pub type Network = BoundedVec<u8, MaxStringLength>;
-pub type AssertionNetworks = BoundedVec<Network, MaxStringLength>;
+pub type AssertionNetworks = BoundedVec<AssertionNetwork, MaxStringLength>;
```
Comment (Kailai-Wang) [inline primitives/core/src/assertion.rs]: I don't think we need `AssertionNetworks` anymore

File: tee-worker/litentry/core/data-providers/src/graphql.rs [hunk 2]
```diff
@@ -83,6 +83,35 @@ impl From<EvmNetwork> for VerifiedCredentialsNetwork {
 	}
 }
 
+impl From<AssertionNetwork> for VerifiedCredentialsNetwork {
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/data-providers/src/graphql.rs]: If I'm allowed to be a bit picky: VC = Verifiable Credential.

So it should be either `VerifiableCredentialNetwork` or `VCNetwork` or maybe `SupportedNetwork`, and please add a comment in the struct definition why we need this (or what's the difference from `AssertionNetwork`).

What's even better is that we have a trait for each data provider, something like `get_supported_network`, which returns a subset of `AssertionNetwork` => but we can do it later

File: tee-worker/ts-tests/vc.test.ts [hunk 3]
```diff
@@ -26,29 +26,29 @@ describeLitentry('VC test', async (context) => {
...
-    });
+    // step('check user sidechain storage before create', async function () {
+    //     const resp_shieldingKey = await checkUserShieldingKeys(
+    //         context,
+    //         'IdentityManagement',
+    //         'UserShieldingKeys',
+    //         u8aToHex(context.substrateWallet.alice.addressRaw)
+    //     );
+    //     assert.equal(resp_shieldingKey, '0x', 'resp_shieldingKey should be empty before set');
+    // });
```
Comment (Kailai-Wang) [inline tee-worker/ts-tests/vc.test.ts]: What's the reason for commenting out these? If it's needed we should have the comment somewhere.

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]

File: tee-worker/ts-tests/vc.test.ts (hunk 3 above)
Comment (outofxxx) [reply tee-worker/ts-tests/vc.test.ts]: Accidentally commented out for local testing, it will be restored

**Comment by Kailai-Wang** [comment]
As mentioned it's OK to leave as it is right now as the refactoring is not desperately needed. But please leave a comment/TODO that we'll refactor out `VerifiedCredentialNetwork` later

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**